From a2c550b445e6fd3d9c944bf889895f61f8f3e68b Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 07 May 2010 15:32:06 +0000
Subject: Fix loading OSD icons when there's no SVG version

In gnome-icon-theme in 2.30.x, there's no SVG icons anymore,
so the OSD popups won't look like the rest of the system,
as "hand-drawn" icons will be used instead.

https://bugzilla.gnome.org/show_bug.cgi?id=618023
---
diff --git a/plugins/media-keys/gsd-media-keys-window.c b/plugins/media-keys/gsd-media-keys-window.c
index 4b736e9..2547068 100644
--- a/plugins/media-keys/gsd-media-keys-window.c
+++ b/plugins/media-keys/gsd-media-keys-window.c
@@ -208,22 +208,9 @@ load_pixbuf (GsdMediaKeysWindow *window,
         pixbuf = gtk_icon_theme_load_icon (theme,
                                            name,
                                            icon_size,
-                                           GTK_ICON_LOOKUP_FORCE_SVG,
+                                           GTK_ICON_LOOKUP_FORCE_SIZE,
                                            NULL);
 
-        /* make sure the pixbuf is close to the requested size
-         * this is necessary because GTK_ICON_LOOKUP_FORCE_SVG
-         * seems to be broken */
-        if (pixbuf != NULL) {
-                int width;
-
-                width = gdk_pixbuf_get_width (pixbuf);
-                if (width < (float)icon_size * 0.75) {
-                        g_object_unref (pixbuf);
-                        pixbuf = NULL;
-                }
-        }
-
         return pixbuf;
 }
 
--
cgit v0.8.3.1
